apps/dokploy/server/utils/builders/nixpacks.ts

```diff
@@ -20,11 +22,56 @@ export const buildNixpacks = async (
 			args.push("--env", env);
 		}
 
+		if (publishDirectory) {
+			/* No need for any start command, since we'll use nginx later on */
+			args.push("--no-error-without-start");
+		}
+
 		await spawnAsync("nixpacks", args, (data) => {
 			if (writeStream.writable) {
 				writeStream.write(data);
 			}
 		});
+
+		/*
+			Run the container with the image created by nixpacks,
+			and copy the artifacts on the host filesystem.
+			Then, remove the container and create a static build.
+		*/
+
+		if (publishDirectory) {
+			await spawnAsync(
+				"docker",
+				["create", "--name", `${appName}-temp`, appName],
+				(data) => {
+					if (writeStream.writable) {
+						writeStream.write(data);
+					}
+				},
+			);
+
+			await spawnAsync(
+				"docker",
+				[
+					"cp",
+					`${appName}-temp:/app/${publishDirectory}`,
+					path.join(buildAppDirectory, publishDirectory),
+				],
+				(data) => {
+					if (writeStream.writable) {
+						writeStream.write(data);
+					}
+				},
+			);
+
+			await spawnAsync("docker", ["rm", `${appName}-temp`], (data) => {
+				if (writeStream.writable) {
+					writeStream.write(data);
+				}
+			});
+
+			await buildStatic(application, writeStream);
+		}
```

Comment:
I see a problem here that could make it impossible to recover from an error.

1. We assign a publish directory example /dist
2. We deploy 
3. This will create a temporary container and try to copy what is in that container based on the publish directory property, if that directory does not exist it will fail at this point and there is no way to recover from that error in any way.
4. The temporary container was never deleted because it never got to execute that line so we will be in a cycle of trying to recreate the temporary container but we will not be able to do it because it still exists.

Definetely is a good idea this, I tested and it decreases from 700 mb to 60mb but I want to think more about the problem and different cases that can have

Reply:
We can wrap the whole if in a try-catch block, and case of error, cancel the temporary container.

As an alternative, I tried yesterday to modify the Nixpacks start plan, using Nginx as runImage.
It was trickier because you had to save the plan into the filesystem, make the changes, and run the image again. Still, the image was around 100 MB because we can't use nginx:alpine.

Coolify creates an image from the initial image and copies the files with a `COPY --from=IMAGE /publish_directory ./`. It definitely works, but you need to keep the initial image; otherwise, you can't rebuild, and you won't save space.

I felt this was the most reliable method, especially because it allows us to build a static image independently from the build phase.

Reply:
@lorenzomigliorero What I meant was that you can't recover from an error with the current code, I gave you the steps so you can analyze that problem.

what you should do would be to first delete the image forcibly in a try catch and then create the temporary container and at the end again delete the container if it exists, this because there may be certain problems between the creation of the temporary container and the copy of directories so it is necessary to forcibly try to delete at the beginning and end

Reply:
Many months ago this was my way to deploy and create temporary containers and in the end I eliminated this way because it gave many problems, but with the solution that I gave you above to eliminate at the beginning and at the end would be enough.

Reply:
I removed the container in the catch block. If the container doesn't exist because the publishDirectory isn't settled, the docker rm command gracefully exists.